src/averageAges.js

```diff
@@ -55,7 +69,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const callBackSon = (sum, child) =>
+    sum + (child.born
+      - people.find(mother => mother.name === child.mother).born);
+
+  const callBackFilter = (child) =>
+    onlyWithSon === true
+      ? people.some(mother => child.mother === mother.name
+          && child.sex === 'm')
+      : people.some(mother => child.mother === mother.name);
```

Comment:
You have boolean type, so, you don't need to do this
```suggestion
  const callBackFilter = (child) =>
    onlyWithSon
      ? people.some(mother => child.mother === mother.name
          && child.sex === 'm')
      : people.some(mother => child.mother === mother.name);
```